lib/ansible/modules/database/postgresql/postgresql_set.py

```diff
@@ -0,0 +1,518 @@
+
+    if not HAS_PSYCOPG2:
+        module.fail_json(msg="the python psycopg2 module is required")
+
+    # To use defaults values, keyword arguments must be absent, so
+    # check which values are empty and don't include in the **kw
+    # dictionary
+    params_map = {
+        "login_host": "host",
+        "login_user": "user",
+        "login_password": "password",
+        "port": "port",
+        "db": "database",
+        "ssl_mode": "sslmode",
+        "ssl_rootcert": "sslrootcert"
+    }
+    kw = dict((params_map[k], v) for (k, v) in iteritems(module.params)
+              if k in params_map and v != '' and v is not None)
+
+    # Store connection parameters for the final check:
+    con_params = deepcopy(kw)
+
+    # If a login_unix_socket is specified, incorporate it here.
+    is_localhost = "host" not in kw or kw["host"] is None or kw["host"] == "localhost"
+    if is_localhost and module.params["login_unix_socket"] != "":
+        kw["host"] = module.params["login_unix_socket"]
+
+    if psycopg2.__version__ < '2.4.3' and sslrootcert:
+        module.fail_json(msg='psycopg2 must be at least 2.4.3 '
+                             'in order to user the ssl_rootcert parameter')
+
+    db_connection = connect_to_db(module, kw, autocommit=True)
+    cursor = db_connection.cursor(cursor_factory=psycopg2.extras.DictCursor)
+
+    # Check server version (needs 9.4 or later):
+    cursor.execute('SELECT version()')
+    ver = cursor.fetchone()[0].split()[1]
+    ver = '.'.join(ver.split('.')[:2])
+    if PG_REQ_VER > float(ver):
+        module.warn("PostgreSQL is %s version but %s or later is required" % (ver, PG_REQ_VER))
+        kw = dict(
+            changed=False,
+            restart_required=False,
+            value_pretty="",
+            prev_val_pretty="",
+            value={"value": "", "unit": ""},
+        )
+        kw['name'] = name
+        db_connection.close()
+        module.exit_json(**kw)
+
+    # Switch role, if specified:
+    if session_role:
+        try:
+            cursor.execute('SET ROLE %s' % session_role)
+        except Exception as e:
+            module.fail_json(msg="Could not switch role: %s" % to_native(e))
+
+    # Set default returned values:
+    restart_required = False
+    changed = False
+    kw['name'] = name
+    kw['restart_required'] = False
+
+    # Get info about param state:
+    res = param_get(cursor, module, name)
+    current_value = res[0]
+    raw_val = res[1]
+    unit = res[2]
+    boot_val = res[3]
+    context = res[4]
+
+    if value == 'True':
+        value = 'on'
+    elif value == 'False':
+        value = 'off'
+
+    kw['prev_val_pretty'] = current_value
+    kw['value_pretty'] = deepcopy(kw['prev_val_pretty'])
+    kw['context'] = context
+
+    # Do job
+    if context == "internal":
+        module.fail_json(msg="%s: cannot be changed (internal context). See documentation" % name)
```

Comment:
I'd add this link https://www.postgresql.org/docs/current/runtime-config-preset.html to get more details

Reply:
It makes sense, I agree